File: VB/SubReportExample.Module/ContactReportWithSubreportMappedToViewDataSource.vb
Imports System
Imports DevExpress.XtraReports.UI
Imports DevExpress.ExpressApp.Utils
Imports DevExpress.ExpressApp

Namespace SubReportExample.Module
    Partial Public Class ContactReportWithSubreportMappedToViewDataSource
        Inherits DevExpress.XtraReports.UI.XtraReport

        Public Sub New()
            InitializeComponent()
        End Sub
        Private Sub xrSubreport1_BeforePrint(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintEventArgs) Handles xrSubreport1.BeforePrint
            Dim subReport As XRSubreport = DirectCast(sender, XRSubreport)
            subReport.ReportSource.Parameters("MasterOid").Value = GetCurrentColumnValue("Oid")
        End Sub
    End Class
End Namespace

File: CS/SubReportExample.Module/ContactReportWithSubreportMappedToViewDataSource.cs
using System;
using DevExpress.XtraReports.UI;
using DevExpress.ExpressApp.Utils;
using DevExpress.ExpressApp;

namespace SubReportExample.Module {
    public partial class ContactReportWithSubreportMappedToViewDataSource : DevExpress.XtraReports.UI.XtraReport {
        public ContactReportWithSubreportMappedToViewDataSource() {
            InitializeComponent();
        }
        private void xrSubreport1_BeforePrint(object sender, System.Drawing.Printing.PrintEventArgs e) {
            XRSubreport subReport = (XRSubreport)sender;
            subReport.ReportSource.Parameters["MasterOid"].Value = GetCurrentColumnValue("Oid");
        }
    }
}
